engine/src/main/java/graphql/nadel/engine/execution/NadelExecutionStrategy.java

```diff
@@ -121,17 +121,44 @@ public NadelExecutionStrategy(List<Service> services,
-            result.add(serviceCF);
+            boolean isNamespaced = !fieldExecutionStepInfo.getFieldDefinition().getDirectives("namespaced").isEmpty();
+            if (isNamespaced) {
+                GraphQLObjectType namespacedObjectType = (GraphQLObjectType) fieldExecutionStepInfo.getUnwrappedNonNullType();
+                for (Map.Entry<Service, Set<GraphQLFieldDefinition>> serviceWithCorrespondingFieldDefinitions : fieldInfos.fieldDefinitionsByService.entrySet()) {
+                    Service service = serviceWithCorrespondingFieldDefinitions.getKey();
+                    Set<GraphQLFieldDefinition> secondLevelFieldDefinitionsForService = serviceWithCorrespondingFieldDefinitions.getValue();
+
+                    Optional<MergedField> newMergedField = MergedFieldUtil.includeSubSelection(mergedField, namespacedObjectType, executionCtx,
+                            field -> secondLevelFieldDefinitionsForService.stream()
+                                    .filter(namespacedObjectType.getFieldDefinitions()::contains)
+                                    .anyMatch(graphQLFieldDefinition -> graphQLFieldDefinition.getName().equals(field.getName())));
+                    if (!newMergedField.isPresent()) {
+                        continue;
+                    }
+                    ExecutionStepInfo newFieldExecutionStepInfo = executionStepInfoFactory.newExecutionStepInfoForSubField(executionCtx, newMergedField.get(), rootExecutionStepInfo);
+                    CreateServiceContextParams parameters = CreateServiceContextParams.newParameters()
+                            .from(executionCtx)
+                            .service(service)
+                            .executionStepInfo(newFieldExecutionStepInfo)
+                            .build();
+
+                    CompletableFuture<Object> serviceContextCF = serviceExecutionHooks.createServiceContext(parameters);
+                    CompletableFuture<OneServiceExecution> serviceCF = serviceContextCF.thenApply(serviceContext -> new OneServiceExecution(service, serviceContext, newFieldExecutionStepInfo));
+                    result.add(serviceCF);
+                }
+            } else {
+                Service service = getServiceForFieldDefinition(fieldExecutionStepInfo.getFieldDefinition());
+
+                CreateServiceContextParams parameters = CreateServiceContextParams.newParameters()
+                        .from(executionCtx)
+                        .service(service)
+                        .executionStepInfo(fieldExecutionStepInfo)
+                        .build();
+
+                CompletableFuture<Object> serviceContextCF = serviceExecutionHooks.createServiceContext(parameters);
+                CompletableFuture<OneServiceExecution> serviceCF = serviceContextCF.thenApply(serviceContext -> new OneServiceExecution(service, serviceContext, fieldExecutionStepInfo));
+                result.add(serviceCF);
+            }
```

Comment:
So much of the following code is the same - dont repeat it.   reactor it so that we take a service and fieldExecutionStepInfo and do this work in one place